options-design/mongo/mongo.go

```diff
@@ -0,0 +1,124 @@
+package mongo
+
+import (
+	"context"
+
+	"github.com/mongodb/mongo-go-driver/options-design/mongo/clientopts"
+	"github.com/mongodb/mongo-go-driver/options-design/mongo/findopts"
+	"github.com/mongodb/mongo-go-driver/options-design/option"
+)
+
+// Client is the main type used to access MongoDB. A client is created from a
+// connection string and can be used to retrieve a Database. To perform
+// individual queries a collection object is retrieved from a Database. For
+// commands that are not supported directly by the porcelain API the RunCommand
+// method on the Database type can be used to execute arbitrary commands
+// against a database.
+type Client struct{}
+
+// NewClient creates a client from the given uri string.
+func NewClient(uri string, opts ...clientopts.Option) (*Client, error) { return nil, nil }
```

Comment:
We had discussed making uri an option as well, so that the ordering of what get's applied first or last is up to the user.

Reply:
Oh right, I forgot about that.